### PR TITLE
Fixes #124: Layout visibility condition for checksums

### LINKED ISSUE #124: Idea: a Layouts visibility condition for CiviCRM checksum
I was just working on something where we send out a checksum link to a private webform, and I was trying to add a block for anonymous users (e.g. "Check your email for the special link") but came to realize even when they click the special link, they will still be anonymous users. Perhaps there's a way to create a CiviCRM-specific visibility condition for Layouts to make sure a valid checksum is in use.

### PR DESCRIPTION
**Pull request by laryn**

### CODE REVIEW

**Comment by laryn** [comment]
Oy. Hopefully that fixes all the coding standards issues. Thank goodness for squash and merge...

File: modules/layouts/plugins/access/civicrm_checksum_layout_access.inc [hunk 1]
```diff
@@ -0,0 +1,123 @@
+<?php
+/**
+ * @file
+ * Plugin to provide access if two user account contexts are equal.
```
Comment (herbdool) [inline modules/layouts/plugins/access/civicrm_checksum_layout_access.inc]: @laryn Can you make this file comment more relevant? Something like "Plugin to provide access if a CiviCRM contact checksum is valid, or if a user is logged in."

Comment (laryn) [reply modules/layouts/plugins/access/civicrm_checksum_layout_access.inc]: Thank you!

**Review by herbdool — APPROVED** [review]
Great!

**Pull request merged**